simppl/cli.py

```diff
@@ -83,15 +89,25 @@ def run(self, argv):
         package = self.command_line_tools[self.tool_name_to_package[tool_name]]
 
         if tool_name in package:
-            self.run_tool(tool_name, package, argv[2:])
-            return 0
+            exit_code = self.__run_tool(tool_name, package, argv[1:])
+            return exit_code
         else:
             raise RuntimeError(f'Could not find tool: {tool_name} \n{self.print_usage()}')
 
+    def __add_tools_from_list(self):
+        for tool in self.modules_list:
+            tool_name = tool.__name__.split('.')[-1]
+            package = tool.__package__
+            self.command_line_tools[package][tool_name] = tool.run
+            self.tool_name_to_package[tool_name] = package
+
     @staticmethod
-    def run_tool(tool_name: str, package: dict, arguments_list):
-        argv = [tool_name] + arguments_list
-        package[tool_name](argv)
+    def __run_tool(tool_name: str, package: dict, argv: List[str]) -> int:
```

Comment:
it won't break the API? was it always used as a private method?

Reply:
yes. I just didn't mark it as private before